fhir-bucket/src/main/java/com/ibm/fhir/bucket/interop/GetExplanationOfBenefitDetail.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.bucket.interop;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.http.HttpStatus;
+
+import com.ibm.fhir.bucket.client.FhirClient;
+import com.ibm.fhir.bucket.client.FhirClientUtil;
+import com.ibm.fhir.bucket.client.FhirServerResponse;
+import com.ibm.fhir.model.resource.Bundle;
+import com.ibm.fhir.model.resource.Bundle.Entry;
+import com.ibm.fhir.model.resource.Bundle.Entry.Request;
+import com.ibm.fhir.model.resource.ExplanationOfBenefit;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.type.Url;
+import com.ibm.fhir.model.type.code.BundleType;
+import com.ibm.fhir.model.type.code.HTTPVerb;
+
+/**
+ * Use the response bundle from the GetPatientBundle request and use it to retrieve
+ * other resources referenced by the ExplanationOfBenefit entries
+ */
+public class GetExplanationOfBenefitDetail {
+    private static final Logger logger = Logger.getLogger(GetPatientBundle.class.getName());
+    
+    // The logical id of the Patient for which we are retrieving information
+    private final Bundle inputBundle;
+        
+    public GetExplanationOfBenefitDetail(Bundle inputBundle) {
+        this.inputBundle = inputBundle;
+    }
+
+    /**
+     * Get the bundle containing the Patient resource plus a list of ExplanationOfBenefit
+     * resources associated with the patient
+     * @param client
+     * @return
+     */
+    public Bundle run(FhirClient client) {
+        Bundle.Builder bundleBuilder = Bundle.builder();
+        
+        // Build a bundle with gets for each ExplanationOfBenefit we can find in the
+        // inputBundle
+        for (Entry entry: inputBundle.getEntry()) {
+            Resource r = entry.getResource();
+            if (r.is(Bundle.class)) {
+                // this should be the result of the ExplanationOfBenefit search, so iterate
+                // over each entry. TODO should visit instead
```

Comment:
If you want to 'visit', the best is FHIRPath

Reply:
I know our convention is not to leave TODO...but I sometimes like to leave notes of better ways to do things as inspiration in case we come back and refactor stuff in the future.

Reply:
oh my comment was on the ability to use a visitor, I didn't have a problem with the todo.